lambda_backend/route_microservice/src/getRoutesByUser.ts

```diff
@@ -0,0 +1,68 @@
+import { Handler } from 'aws-lambda';
+import DynamoDB, { AttributeValue, QueryInput } from 'aws-sdk/clients/dynamodb';
+import createError from 'http-errors';
+
+import { getMiddlewareAddedHandler } from './common/middleware';
+import { getRoutesByUserSchema } from './common/schema';
+import { GetRoutesByUserEvent, GymItem, UserRoutesIndexItem } from './common/types';
+
+const dynamoDb = new DynamoDB.DocumentClient();
+
+const getRoutesByUser: Handler = async (event: GetRoutesByUserEvent) => {
+  if (!process.env['ROUTE_TABLE_NAME'] || !process.env['GYM_TABLE_NAME']) {
+    throw createError(500, 'Route or Gym table name is not set');
+  }
+  const {
+    queryStringParameters: { username },
+  } = event;
+  const routeQueryInput: QueryInput = {
+    TableName: process.env['ROUTE_TABLE_NAME'],
+    IndexName: 'userRoutesIndex',
+    ConsistentRead: false,
+    ScanIndexForward: false,
+    KeyConditionExpression: 'username = :username',
+    ExpressionAttributeValues: {
+      ':username': username as AttributeValue,
+    },
+  };
+  let userRouteIndexItems: UserRoutesIndexItem[];
+  try {
+    const response = await dynamoDb.query(routeQueryInput).promise();
+    userRouteIndexItems = response.Items as UserRoutesIndexItem[];
+  } catch (error) {
+    throw createError(500, 'Error querying table', error);
+  }
+
+  // Assumes that one user will not post from too many countries
```

Comment:
What happens if they do?

Reply:
Each query by country should be 0.5 RCU, I put max 3 RCU for gym data table. So if they post from more than 6 countries, there might be a problem. But dirty way to resolve is every 6 countries, put a 1 sec delay.